src/vm/corhost.cpp

```diff
@@ -727,56 +726,95 @@ HRESULT CorHost2::_CreateAppDomain(
-        
-        if (nProperties>0)
+        ARG_SLOT args[3];
+        args[0] = PtrToArgSlot(pPropertyNames);
+        args[1] = PtrToArgSlot(pPropertyValues);
+        args[2] = PtrToArgSlot(nProperties);
+
+        setup.Call(args);
+    }
+
+    LPCWSTR pwzNativeDllSearchDirectories = NULL;
+    LPCWSTR pwzTrustedPlatformAssemblies = NULL;
+    LPCWSTR pwzPlatformResourceRoots = NULL;
+    LPCWSTR pwzAppPaths = NULL;
+    LPCWSTR pwzAppNiPaths = NULL;
+#ifdef FEATURE_COMINTEROP
+    LPCWSTR pwzAppLocalWinMD = NULL;
+#endif
+
+    for (int i = 0; i < nProperties; i++)
+    {
+        if (wcscmp(pPropertyNames[i], W("NATIVE_DLL_SEARCH_DIRECTORIES")) == 0)
         {
-            _gc.propertyNames = (PTRARRAYREF) AllocateObjectArray(nProperties, g_pStringClass);
-            _gc.propertyValues= (PTRARRAYREF) AllocateObjectArray(nProperties, g_pStringClass);
-            for (int i=0;i< nProperties;i++)
-            {
-                STRINGREF obj = StringObject::NewString(pPropertyNames[i]);
-                _gc.propertyNames->SetAt(i, obj);
-                
-                obj = StringObject::NewString(pPropertyValues[i]);
-                _gc.propertyValues->SetAt(i, obj);
-            }
+            pwzNativeDllSearchDirectories = pPropertyValues[i];
         }
+        else
+        if (wcscmp(pPropertyNames[i], W("TRUSTED_PLATFORM_ASSEMBLIES")) == 0)
+        {
+            pwzTrustedPlatformAssemblies = pPropertyValues[i];
+        }
+        else
+        if (wcscmp(pPropertyNames[i], W("PLATFORM_RESOURCE_ROOTS")) == 0)
+        {
+            pwzPlatformResourceRoots = pPropertyValues[i];
+        }
+        else
+        if (wcscmp(pPropertyNames[i], W("APP_PATHS")) == 0)
+        {
+            pwzAppPaths = pPropertyValues[i];
+        }
+        else
+        if (wcscmp(pPropertyNames[i], W("APP_NI_PATHS")) == 0)
+        {
+            pwzAppNiPaths = pPropertyValues[i];
+        }
+#ifdef FEATURE_COMINTEROP
+        else
+        if (wcscmp(pPropertyNames[i], W("APP_LOCAL_WINMETADATA")) == 0)
+        {
+            pwzAppLocalWinMD = pPropertyValues[i];
+        }
+#endif
+    }
 
-        MethodDescCallSite setup(METHOD__APP_DOMAIN__SETUP);
-
-        ARG_SLOT args[3];
-        args[0]=ObjToArgSlot(_gc.friendlyName);
-        args[1]=ObjToArgSlot(_gc.propertyNames);
-        args[2]=ObjToArgSlot(_gc.propertyValues);
+    pDomain->SetNativeDllSearchDirectories(pwzNativeDllSearchDirectories);
 
-        setup.Call(args);
-        
-        GCPROTECT_END();
+    {
+        SString sTrustedPlatformAssemblies(pwzTrustedPlatformAssemblies);
+        SString sPlatformResourceRoots(pwzPlatformResourceRoots);
+        SString sAppPaths(pwzAppPaths);
+        SString sAppNiPaths(pwzAppNiPaths);
 
-        *pAppDomainID=pDomain->GetId().m_dwId;
+        CLRPrivBinderCoreCLR *pBinder = pDomain->GetTPABinderContext();
+        _ASSERTE(pBinder != NULL);
+        IfFailThrow(pBinder->SetupBindingPaths(
```

Comment:
Technically before your change this was only called if the `pwzTrustedPlatformAssemblies` was specified. I think it was basically wrong before, since if we didn't call `SetupBindingPaths` it seems like things would break. So I think this change is OK.